### PR TITLE
llvm.lround: Added the intrinsic cost model case with a test.

### PR DESCRIPTION
**Pull request by sgundapa**

### CODE REVIEW

File: llvm/test/Analysis/CostModel/AMDGPU/lround.ll [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+; RUN: opt -passes="print<cost-model>" 2>&1 -disable-output -mtriple=amdgcn-unknown-amdhsa < %s | FileCheck -check-prefixes=ALL %s
+; RUN: opt -passes="print<cost-model>" -cost-kind=code-size 2>&1 -disable-output -mtriple=amdgcn-unknown-amdhsa -mcpu=gfx1010 < %s | FileCheck -check-prefixes=SIZE %s
+; RUN: opt -passes="print<cost-model>" -cost-kind=code-size 2>&1 -disable-output -mtriple=amdgcn-unknown-amdhsa -mcpu=gfx90a < %s | FileCheck -check-prefixes=SIZE %s
+; RUN: opt -passes="print<cost-model>" -cost-kind=code-size 2>&1 -disable-output -mtriple=amdgcn-unknown-amdhsa -mcpu=gfx900 < %s | FileCheck -check-prefixes=SIZE %s
+; RUN: opt -passes="print<cost-model>" -cost-kind=code-size 2>&1 -disable-output -mtriple=amdgcn-unknown-amdhsa < %s | FileCheck -check-prefixes=SIZE %s
+
+
+define i32 @test_lround() {
+; ALL-LABEL: 'test_lround'
+; ALL-NEXT:  Cost Model: Found an estimated cost of 1 for instruction: %VI32F32 = call i32 @llvm.lround.i32.f32(float undef)
```
Comment (arsenm) [inline llvm/test/Analysis/CostModel/AMDGPU/lround.ll:14]: This should probably be 2; I believe this codegens to 2 full rate instructions

**Review by arsenm — COMMENTED** [review]
Can you also add an aarch64 or x86 testcase? The baseline cost also seems too low

File: llvm/test/Analysis/CostModel/AMDGPU/lround.ll (hunk 1 above)
Comment (sgundapa) [reply llvm/test/Analysis/CostModel/AMDGPU/lround.ll:14]: Thats right, FPTOSI and ROUND. I will fix it.

**Comment by arsenm** [comment]
Why was this closed?

**Pull request closed without merging**